### ISSUE TITLE
Add support to set server URL in tenant config

### ISSUE THREAD

**Issue by MyIgel** [minio/operator#858]
**Is your feature request related to a problem? Please describe.**
The URL can be set via the `MINIO_SERVER_URL` env variable which can currently only be set when overriding the whole env config https://github.com/minio/operator/issues/790#issuecomment-917476406 which has sideeffects (`User is not allowed to be same as admin access key` error).

**Describe the solution you'd like**
It would be reallly useful to allow setting the server URL in the tenant config.

**Describe alternatives you've considered**
Above workaround has sideeffects so its more a hack that might work but cant imho be considered an alternative, esp. as I can't imagine that having buckets exposed under a public facing URL is such a niche use case that its not supported to configure it in an easy and useful way.

**Additional context**
_none_

**Comment by GaruGaru** [comment]
Any update on this ?

**Comment by harshavardhana** [comment]
You can change MINIO_SERVER_URL using `config.env` support to override whatever settings you want.

**Comment by chengjianhua** [comment]
I have the same requirement.

I'm using https://github.com/minio/operator to deploy MinIO into the Kubernetes cluster and using the CRD `tenant` to configure the tenant.

I followed the guide mentioned in https://github.com/minio/operator/issues/790#issuecomment-917476406,  and created a secret like below:

```yaml
apiVersion: v1
stringData:
  config.env: |
    export MINIO_ROOT_USER="root_user"
    export MINIO_ROOT_PASSWORD="Rt4XfNT3Fr8xRuyU"
    export MINIO_SERVER_URL="http://myexample.com"
kind: Secret
metadata:
  name: minio-tenant-config
```

After configuring the CRD `tenant` to has `configuretion:\n  name: minio-tenant-config`, and all new pods were ready, I failed to log in to the console with the above user and password. It responded with 403.

After a while, I recognized that I should put all necessary environment variables into `config.env`, so I added more environment variables that have already existed in the pod's `env` field into the secret `minio-tenant-config`, so it looks like as below finally:

```yaml
apiVersion: v1
stringData:
  config.env: |
    export MINIO_ROOT_USER="root_user"
    export MINIO_ROOT_PASSWORD="Rt4XfNT3Fr8xRuyU"
    export MINIO_SERVER_URL="http://myexample.com"
    export MINIO_ENDPOINTS="http://console-dev-minio-tenant-ss-0-{0...3}.console-dev-minio-tenant-hl.mars-web-dev.svc.cluster.local/export{0...3}/data"
    export MINIO_UPDATE="on"
    export MINIO_UPDATE_MINISIGN_PUBKEY="RWTx5Zr1tiHQLwG9keckT0c45M3AGeHD6IvimQHpyRywVWGbP1aVSGav"
    export MINIO_OPERATOR_VERSION="v4.3.7"
    export MINIO_ARGS="env+tls://K3dqKcKTzGdVuHitPxwZ:GBqhXKUSwOG3Elmsg2n3kjH3oVFvMwdJumTb2Q3S@operator.mars-web-dev.svc.cluster.local:4222/webhook/v1/getenv/mars-web-dev/console-dev-minio-tenant"
kind: Secret
metadata:
  name: console-dev-minio-tenant-config
```

The drawback of this approach is that I have to put a variable like `MINIO_ENDPOINTS="http://console-dev-minio-tenant-ss-0-{0...3}.console-dev-minio-tenant-hl.mars-web-dev.svc.cluster.local/export{0...3}/data"` into the secret, the pattern of this value seems should be keep internally by operator itself...

In other words, I think it's improper for users (like me) to generate a value like this.

**I'm afraid the pattern of these values would change, and then I have to catch up with the changes manually.**

@harshavardhana

**Comment by chengjianhua** [comment]
This requirement requires few changes to the operator's implementation, only need to change the order of appending user-specified environment variables:
https://github.com/minio/operator/blob/ede9e5936c281cc2078566693abef226af5f4a88/pkg/resources/statefulsets/minio-statefulset.go#L76

The current order is, putting user-specified environment variables first, then appending operator-computed environment variables later.

If the operator could reverse the above order, overriding default environment variables is possible.

@harshavardhana What do you think about this?

**Comment by chengjianhua** [comment]
> The drawback of this approach is that I have to put a variable like MINIO_ENDPOINTS="http://console-dev-minio-tenant-ss-0-{0...3}.console-dev-minio-tenant-hl.mars-web-dev.svc.cluster.local/export{0...3}/data" into the secret, the pattern of this value seems should be keep internally by operator itself...

Embarrassed...I just found that after another re-creation of tenant CRD, I noticed that `MINO_ENDPOINTS` is injected in `containers[*].env`...
It's wired that I used to don't see it before...

----

But I still request that it's allowed to override `MINIO_SERVER_URL` in the `spec.env` directly, rather than configuring them in another secret.

**Comment by harshavardhana** [comment]
> But I still request that it's allowed to override `MINIO_SERVER_URL` in the `spec.env` directly, rather than configuring them in another secret.

Another secret is provided on purpose to make as many changes as you want. Providing it through multiple ways is not necessary and required at this point.
